llama_stack/apis/providers/providers.py

```diff
@@ -37,4 +33,4 @@ class Providers(Protocol):
     async def list_providers(self) -> ListProvidersResponse: ...
 
     @webmethod(route="/providers/{provider_id}", method="GET")
-    async def inspect_provider(self, provider_id: str) -> GetProviderResponse: ...
+    async def inspect_provider(self, provider_id: str) -> Optional[ProviderInfo]: ...
```

Comment:
shouldn't be optional I think. We should return 4xx if it doesn't exist.

Reply:
Yeah I was considering this. Or I guess the client could handle it

Reply:
+1 @booxter that's the correct REST behavior

Reply:
This should be applicable for all GET endpoints, and we should fix it altogether at once. Filed an issue: https://github.com/meta-llama/llama-stack/issues/1630

Reply:
fixed it for this one

Reply:
@leseb has a great fix that makes all GETs behave consistently (with enforcement through API generator): https://github.com/meta-llama/llama-stack/pull/1635 I'd love to see this merging ASAP so that we can start adding more enforcement rules (I have a number in mind). Please review!